activerecord/lib/active_record/connection_handling.rb

```diff
@@ -377,5 +379,11 @@ def swap_connection_handler(handler, &blk) # :nodoc:
       ensure
         ActiveRecord::Base.connection_handler = old_handler
       end
+
+      def check_own_connection_established(klass)
+        unless klass.name == klass.connection_specification_name
+          raise NotImplementedError, "calling `connected_to` is only allowed on abstract class that established the connection."
```

Comment:
I don't think this gives the user enough information. There's a big difference between calling `connected_to` on a non-abstract class and calling it on the connection that doesn't have an established connection. It's also confusing that `connected_to` and `connected_to_many` handle this error differently. It should be handled the same. Can you update this to handle those cases separately and reinstate the previous error message when it's not an abstract class? 

Also there's a `the` missing in `is only allowed on abstract class` should be `is only allowed on the abstract class`.